pkg/ngrokapidriver/driver.go

```diff
@@ -183,28 +191,24 @@ func (nc ngrokAPIDriver) GetReservedDomains(ctx context.Context, edgeID string)
 	}
 	hostPortDomains := []string{}
 	for _, hostport := range *edge.Hostports {
-		hostPortDomains = append(hostPortDomains, strings.Split(hostport, ":")[0])
+		domain, _, err := net.SplitHostPort(hostport)
+		if err != nil {
+			return nil, err
+		}
+		hostPortDomains = append(hostPortDomains, domain)
 	}
 
 	domainsItr := nc.reservedDomains.List(nil)
 	var matchingReservedDomains []ngrok.ReservedDomain
 	// Loop while there are more domains and check if they match any of the hostPortDomains. If so add it to the reservedDomains
-	for {
-		if !domainsItr.Next(ctx) {
-			err := domainsItr.Err()
-			if err != nil {
-				return nil, err
-			}
-			break
-		}
+	for domainsItr.Next(ctx) {
```

Comment:
this `for` construct, and then doing a single `if iter.Err() != nil` after the loop, is the more idiomatic way to use an iterator than relying on a `break` to escape a `while true` loop.